rust/eos-rs/src/api/dungeon_mode/entity.rs

```diff
@@ -0,0 +1,117 @@
+use crate::api::dungeon_mode::*;
+use crate::api::objects::*;
+
+/// Extension trait for [`DungeonEntity`].
+///
+/// # Important safety note
+/// Implementations of this trait can assume that overlay 29 is loaded (since this is the
+/// only context dungeon entities are actually relevant). If you manually implement this trait,
+/// for some reason, you NEED to make sure overlay 29 is loaded when using some functions of this
+/// trait.
+/// If you use the DungeonEntity structs manually outside of dungeon mode, this trait will be
+/// unsafe to use.
+/// The trait and its functions are marked safe for convenience, since in it's intended use case,
+/// overlay 29 will always be loaded.
+pub trait DungeonEntityExt {
```

Comment:
I didn't want to write impl blocks directly for types from the ffi module, but yeah, you probably never need to implement them yourself. Idk, I still feel like this seperation of actual game code and Rust behavior isn't too bad to have here.
What do the others think?

Reply:
Hmm yeah I guess it is nicer to separate these impls into their own files, and I guess there's no better way to do it (without making a wrapper struct) other than with a trait like this. Darn orphan rule.

Yeah I think this is fine. I was just confused why it was done this way, but it makes sense.

Reply:
That's actually not true. Since it's all in the same crate I could make an impl for the structs directly. 
Yeah maybe I will, I'll do some research. I've seen this Ext-trait pattern a few times in libraries that wrap a C API, but I'll check why they did that.

One thing I'm concerned about is documentation. I think even if I make the impl for the DungeonEntity type alias, rustdoc would not show it and only show it for ffi::entity, which would be kinda really bad in terms of documentation, since then you'd only find it by digging through the ffi module's documentation.

Reply:
Regarding the documentation issue, you could re-export the ffi type aliases to make them easier to find?

If the Ext-trait pattern is widely used though, I assume it has some merits. Again, I don't think it's inherently bad.

Reply:
> Regarding the documentation issue, you could re-export the ffi type aliases to make them easier to find?

Yeah like I said, rustdoc won't show the trait implementations for alias types. So I think it would be kinda hard to find in the documentation. I think I'll keep this as a seperate trait, for the sake of clarity, but maybe I'll seal it so it can't be implemented outside of the crate.